### ISSUE TITLE
Diff-awareness crashes on added submodules 

### ISSUE THREAD

**Issue by underyx** [returntocorp/bento#336]
https://github.com/returntocorp/semgrep/pull/1127/checks?check_run_id=804559293

```
A  ocaml-tree-sitter
```

bento’s diff awareness sees this in the git diff output, and assume it means that file was added

seems like we didn’t anticipate that submodules show up the same way in git diff as new files

one way to fix it would be to always add a `path.is_file()` guard later, but a better solution would be to find a git flag that excludes changed submodules from the output

**Comment by dlukeomalley** [comment]
@underyx Is this issue present in Semgrep and/or something we need to consider as we port diff-awareness?

**Comment by underyx** [comment]
Yes, this is an easy mistake to make when implementing diff-awareness.
